include/zephyr/pm/state.h

```diff
@@ -154,6 +155,15 @@ struct pm_state_info {
 	 * @note 0 means that this property is not available for this state.
 	 */
 	uint32_t exit_latency_us;
+
+	/**
+	 * Whether or not this state triggers device power management.
+	 *
+	 * When this property is true the power management subsystem
+	 * will suspend devices before entering this state and will
+	 * properly resume them when leaving it.
+	 */
+	bool pm_device_enabled;
```

Comment:
Potentially memory usage could be optimised by placing this with `uint8_t substate_id;` to reduce padding

Reply:
yep, that makes sense, this way  we just use the space in the current padding,